src/main/java/semantics/RDFDatasetDirectStatementDeleter.java

```diff
@@ -0,0 +1,349 @@
+package semantics;
+
+import static semantics.RDFImport.RELATIONSHIP;
+import static semantics.RDFParserConfig.URL_SHORTEN;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader.InvalidCacheLoadException;
+import com.google.common.collect.Iterators;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import org.eclipse.rdf4j.model.BNode;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.rio.RDFHandlerException;
+import org.neo4j.graphdb.Direction;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Relationship;
+import org.neo4j.graphdb.RelationshipType;
+import org.neo4j.graphdb.Result;
+import org.neo4j.helpers.collection.Iterables;
+import org.neo4j.logging.Log;
+
+/**
+ * This class implements an RDF handler to statement-wise delete imported RDF data sets
+ *
+ * Created on 18/06/2019.
+ *
+ * @author Emre Arkan
+ */
+
+class RDFDatasetDirectStatementDeleter extends RDFDatasetToLPGStatementProcessor implements
+    Callable<Integer> {
+
+  private static final Label RESOURCE = Label.label("Resource");
+
+  private Cache<ContextResource, Node> nodeCache;
+  private long notDeletedStatementCount;
+  private long bNodeCount;
+  private String bNodeInfo;
+
+  RDFDatasetDirectStatementDeleter(GraphDatabaseService db, RDFParserConfig conf, Log l) {
+    super(db, conf);
+    nodeCache = CacheBuilder.newBuilder()
+        .maximumSize(conf.getNodeCacheSize())
+        .build();
+    log = l;
```

Comment:
This should be part of the super-constructor. Relying on the sub-classes to initialize this value is bad design.

Should be refactored, so that the constructor of `RDFDatasetToLPGStatementProcessor` requires a `Log` and sets its value on construction (ideally as `final`)